tests/FSharp.Compiler.ComponentTests/ErrorMessages/TypeMismatchTests.fs

```diff
@@ -7,6 +7,88 @@ open FSharp.Test.Compiler
 
 module ``Type Mismatch`` =
 
+    module ``Different tuple lengths`` =
+
+        [<Fact>]
+        let ``Known type on the left``() =
+            FSharp """
+let x a b c : int * int = a, b, c
```

Comment:
Could one of the parameters also have a type annotation, to see if it's properly propagated to the right side?

Reply:
Sure, added a test for that.